kotori/daq/strategy/wan.py

```diff
@@ -40,14 +41,69 @@ def topic_to_topology(self, topic):
 
             - "node" is the node identifier. Choose anything you like. This usually
               gets transmitted from an embedded device node.
+
+        Other than decoding the classic WAN path topic style, like
+
+            mqttkit-1/network/gateway/node
+
+        the decoder now also knows how to handle per-device addressing schemes, like
+
+            mqttkit-1/d/123e4567-e89b-12d3-a456-426614174000
+
+        Also, it has another special topic decoding scheme, by decomposing a dashed
+        identifier, and transforming it into the quadruple hierarchy.
+
+            mqttkit-1/dt/network-gateway-node
+
         """
 
         # Decode the topic.
-        m = self.matcher.match(topic)
+        address = None
+
+        # Try to match the per-device pattern.
+        m = self.device_matcher_generic.match(topic)
         if m:
             address = SmartMunch(m.groupdict())
-        else:
-            address = {}
+            if "device" in address:
+                address.network = "devices"
+                address.gateway = "default"
+                address.node = address.device
+                del address.device
+
+        # Try to match the per-device pattern with dashed topology encoding for topics.
+        if address is None:
+            m = self.device_matcher_dashed_topo.match(topic)
+            if m:
+                address = SmartMunch(m.groupdict())
+                if "device" in address:
+                    segments = address.device.split("-")
+
+                    # Compensate "too few segments": Fill up with "default" at the front.
+                    missing_segments = 3 - len(segments)
+                    segments = ["default"] * missing_segments + segments
+
+                    # When there are "too many segments", but the first one matches the realm, we
+                    # assume a slight misconfiguration, and ignore the first decoded segment completely.
+                    if len(segments) > 3:
+                        if segments[0] == address.realm:
+                            segments = segments[1:]
```

Comment:
FIXME: This rule should always be applied, not only when the segments are too many.